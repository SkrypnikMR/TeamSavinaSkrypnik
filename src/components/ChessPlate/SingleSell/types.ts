export type TSingleSell = {
    id: number;
};
