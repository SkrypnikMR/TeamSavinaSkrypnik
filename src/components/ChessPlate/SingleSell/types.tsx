export interface IDiv {
    width: string;
    height: string;
    background: string;
    cursor: string;
}

export type TSingleSell = {
    id: number;
}
