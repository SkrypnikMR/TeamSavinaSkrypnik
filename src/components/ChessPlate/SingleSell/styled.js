import styled from 'styled-components';

export const StSingleSell = styled.div`
    width: 50px;
    height: 50px;
    background: red;
    cursor: pointer;
`;
