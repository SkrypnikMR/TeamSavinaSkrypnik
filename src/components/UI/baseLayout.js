// import styled from 'styled-components';

export const backgroundUrls = {
  light: './public/assets/images/lightTheme.jpg',
  dark: './public/assets/images/darkTheme.jpg',
};
// default color for all transparent backgrounds, use this :)
export const TRANSPARENT_BACKGROUND = 'rgba(0, 0, 0, 0.4)';
//Button
export const bgColorDefault = '#5523C1';
export const colorDefault = 'white';
export const focusColorDefault = 'black';
export const disabledColor = '#666666';
export const bgColorDisabledDefault = '#cccccc';
export const disabledBorder = '1px solid #999999';

//Input
export const bgInpColorDefault = '#F6F7FA';
export const bgFocusColorDefault = '#716f6f';
export const focusColorDefaultInput = 'white';
export const colorDefaultInput = 'black';

export const bgColorDefaultFon = 'rgba(0, 0, 0, 0.4)';
export const bgMessage = 'rgba(196, 196, 196, 0.5)';
export const textColorBlack = 'black';
