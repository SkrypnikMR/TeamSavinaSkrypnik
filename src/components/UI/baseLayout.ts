// default color for all transparent backgrounds, use this :)
export const TRANSPARENT_BACKGROUND = 'rgba(0, 0, 0, 0.4)';
//Button
export const bgColorDefault = '#23EFFC';
export const colorDefault = 'white';
export const focusColorDefault = '#26FDF0';
export const disabledColor = '#666666';
export const bgColorDisabledDefault = '#cccccc';
export const disabledBorder = '1px solid #999999';

//Input
export const bgInpColorDefault = '#F6F7FA';
export const bgFocusColorDefault = '#716f6f';
export const focusColorDefaultInput = 'white';
export const colorDefaultInput = 'black';

export const bgColorDefaultFon = 'rgba(0, 0, 0, 0.4)';
export const bgMessage = 'rgba(196, 196, 196, 0.5)';
export const textColorBlack = 'black';

//Backgrounds
export const darkGrey = '#404040';
export const midGrey = '#808080';
export const lightGray = '#C4C4C4';
export const white = '#ffffff';
export const turquous = '#23EFFC';
export const blackPic = 'rgba(0, 0, 0)';

export const colors = {
  light: {
    background: '#23EFFC',
    backgroundSecondary: '#e5fdff',
    backgroundLogReg: '#e5fdff',
    backgroundItem: '#d7d8e2',
    button: '#2B5278',
    buttonColor: '#fff',
    buttonHover: '#346392',
    input: '#D1D1D1',
    textColor: '#687780',
    buttonText: '#404040',
    placeholderColor: '#55667B',
    AddRoomBtnBtn: '#fff',
  },
  dark: {
    background: '#404040',
    backgroundSecondary: '#C4C4C4',
    backgroundLogReg: '#404040',
    backgroundItem: '#25323f',
    button: '#2B5278',
    buttonColor: '#fff',
    buttonHover: '#346392',
    input: '#242F3D',
    textColor: 'white',
    buttonText: `${bgColorDefault}`,
    placeholderColor: '#55667B',
    AddRoomBtnBtn: '#fff',
  },
};
//MainPage
export const MAIN_PAGE_BG = '#404040';

//RoomContainer
export const ROOMS_CONTAINER_BG = '#C4C4C4';

//SingleRoom
export const SINGLE_ROOM_BG = '#808080';
