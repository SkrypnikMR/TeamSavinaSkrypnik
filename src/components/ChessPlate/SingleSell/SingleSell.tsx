import React from 'react';
import { StSingleSell } from './styled';
import { TSingleSell } from './types';

const SingleSell = ({ id } : TSingleSell) => {
    const handleClick = (e) => {
        console.log(e.target.id);
    };

    return (
        <StSingleSell
            id={id + 1}
            style={{
                background: parseInt((id / 8) + id) % 2 === 0
                    ? 'black' : 'white',
            }}
onClick={handleClick}
        >
{id + 1}
        </StSingleSell>
    );
};

export default SingleSell;
