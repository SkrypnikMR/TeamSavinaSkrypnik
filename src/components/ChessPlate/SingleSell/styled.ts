import styled from 'styled-components';

import { IDiv } from './types';

export const StSingleSell = styled.div <{ backgroungColor?: string }> `
    width: 50px;
    height: 50px;
    cursor: pointer;
    background-color: ${({ backgroungColor }) => backgroungColor};
`;
