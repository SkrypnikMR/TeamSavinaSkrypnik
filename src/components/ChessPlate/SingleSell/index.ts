import SingleSell from './SingleSell';

export default SingleSell;
