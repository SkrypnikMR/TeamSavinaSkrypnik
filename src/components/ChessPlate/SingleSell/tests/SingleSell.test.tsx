import React from 'react';
import { shallow, mount } from 'enzyme';
import { blackPic, white, WHITECHECKER, BLACKCHECKER } from '../../../UI/baseLayout';

import SingleSell from '../SingleSell';

describe('SingleSell', () => {
    let props;
    beforeEach(() => {
        props = {
            id: 1,
            status: { checker: null, blackSquare: false },
            getPosibleStep: jest.fn(),
            position: null,
            doCheckerStep: jest.fn(),
        };
    });
    it('Should match snapshot', () => {
        const component = shallow(<SingleSell {...props}/>);
        expect(component.html()).toMatchSnapshot();
    });
    it('Should render StSingleSell with blackPic', () => {
        const component = mount(<SingleSell {...props}/>);
        expect(component.find('styled__StSingleSell')).toHaveLength(1);
        expect(component.find('styled__StSingleSell').props().color).toBe(white);
    });
    it('Should render StSingleSell with white', () => {
        props.status.blackSquare = true;
        const component = mount(<SingleSell {...props}/>);
        expect(component.find('styled__StSingleSell')).toHaveLength(1);
        expect(component.find('styled__StSingleSell').props().color).toBe(blackPic);
    });
    it('Should render StPossition', () => {
        props.position = true;
        const component = mount(<SingleSell {...props}/>);
        expect(component.find('styled__StPossition')).toHaveLength(1);
    });
    it('Should render img with src = WHITECHECKER', () => {
        props.status.checker = { blackChecker: true };
        const component = mount(<SingleSell {...props}/>);
        expect(component.find('img').props().src).toBe(WHITECHECKER);
    });
    it('Should render img with src = BLACKCHECKER', () => {
        props.status.checker = { blackChecker: false };
        const component = mount(<SingleSell {...props}/>);
        expect(component.find('img').props().src).toBe(BLACKCHECKER);
    });
    it('Should call getPosibleStep', () => {
        props.status.checker = { blackChecker: false };
        const component = mount(<SingleSell {...props}/>);
        (component.find('styled__StSingleSell').props().onClick({ target: { id: 1 } }));
        expect(props.getPosibleStep).toHaveBeenCalledWith(props.id);
    });
    it('Should call doCheckerStep', () => {
        props.position = true;
        const component = mount(<SingleSell {...props}/>);
        (component.find('styled__StSingleSell').props().onClick({ target: { id: 1 } }));
        expect(props.doCheckerStep).toHaveBeenCalledWith(props.id);
    });
});
