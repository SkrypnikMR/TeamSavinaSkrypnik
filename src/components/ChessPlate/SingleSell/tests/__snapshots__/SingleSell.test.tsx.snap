// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleSell Should match snapshot 1`] = `"<div id=\\"NaN\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1rn7xd3-0 iBlbrM\\">NaN</div>"`;
